pkg/apis/build/v1alpha1/image_builds.go

```diff
@@ -57,8 +58,15 @@ func (im *Image) buildNeeded(lastBuild *Build, sourceResolver *SourceResolver, b
 		reasons = append(reasons, BuildReasonBuildpack)
 	}
 
-	if lastBuild.Status.RunImage != "" {
-		lastBuildRunImageRef, err := name.ParseReference(lastBuild.Status.RunImage)
+	if lastBuild.Status.Stack.ID != "" {
```

Comment:
Ideally, buildNeeded should not change if the build is able to "make the optimization" decision.

Reply:
That is true. The run image SHAs will definitely be different.